clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -162,6 +188,19 @@ public synchronized void update(Cluster cluster, long now) {
         this.lastSuccessfulRefreshMs = now;
         this.version += 1;
 
+        // Handle expiry of topics from the metadata list
+        Iterator<Map.Entry<String, Long>> it = topics.entrySet().iterator();
+        while (it.hasNext()) {
+            Map.Entry<String, Long> entry = it.next();
+            Long expireMs = entry.getValue();
+            if (expireMs == TOPIC_EXPIRY_NEEDS_UPDATE)
+                entry.setValue(now + metadataExpireMs);
```

Comment:
It seems a little surprising to reuse `metadataExpireMs` for topic expiration. Users tune this to control how quickly partition changes are detected, which doesn't seem directly related to how often a topic's metadata is actually used. I guess the only downside of early expiration is that we have to fetch metadata sooner than would otherwise be necessary. But since the point is just to protect against unbounded cache growth, I wonder if a suitably large hard-coded value would make more sense? At the very least, a comment on why this makes sense would be helpful.